tests/ConfigControllerTest.php

```diff
@@ -19,31 +34,137 @@ public function testConfigurationIsSavedWithEmptyValues()
+            'context' => 'stores/test123',
+            'shopkey' => '123test',
+            'active_status' => null
+        ]);
+        $configController->handleConfiguration($request);
+
+        /** @var Store $store */
+        $store = Store::where('context', 'stores/test123')->first();
+        $config = Config::where('store_id', $store['id'])->first();
+
+        $this->assertFalse(boolval($config['active']));
+        $this->assertEquals($store['id'], (int) $config['store_id']);
+        $this->assertSame('123test', $config['shopkey']);
+
+        $this->assertEquals(0, $store->scripts->count());
+    }
+
+    public function testSavingNewConfigWillUpdateTheScriptWithNewConfiguration()
+    {
+        $existingStore = new Store([
+            'context' => 'stores/test-shop',
+            'access_token' => 'test-token'
+        ]);
+        $existingStore->save();
+
+        $existingScript = new Script([
+            'name' => 'Findologic',
+            'uuid' => 'test-uuid',
+            'store_hash' => 'test123',
+            'store_id' => $existingStore->id
+        ]);
+        $existingScript->save();
+
+        $mockedStoresXmlResponse = $this->getMockResponse('/xml_response/v2_stores.xml');
+        $mockedScriptsJsonResponse = $this->getMockResponse('json_response/v3_content_scripts.json');
+
+        $mock = new MockHandler([
+            new Response(200, ['Content-Type' => 'application/xml'], $mockedStoresXmlResponse),
+            new Response(204), // Script delete call stores/<store_hash>/v3/content/scripts/<uuid>
+            new Response(200, ['Content-Type' => 'application/json'], $mockedScriptsJsonResponse),
+        ]);
+        $handler = HandlerStack::create($mock);
+        $httpClientMock = new Client(['handler' => $handler]);
+        $configController = new ConfigController($httpClientMock);
+
+        $request = Request::create('/config', 'POST', [
+            'store_hash' => 'test123',
+            'access_token' => 'testAcessToken123',
+            'context' => 'stores/test123',
+            'shopkey' => 'UPDATED_SHOPKEY',
+            'active_status' => 'on'
+        ]);
+        $configController->handleConfiguration($request);
+
+        /** @var Store $store */
+        $store = Store::where('context', 'stores/test123')->first();
+        $config = Config::where('store_id', $store['id'])->first();
+
+        $this->assertTrue(boolval($config['active']));
+        $this->assertEquals($store['id'], (int) $config['store_id']);
+        $this->assertSame('UPDATED_SHOPKEY', $config['shopkey']);
+
+        $this->assertEquals(1, $store->scripts->count());
+        $script = $store->scripts()->first();
+
+        $this->assertEquals('Findologic', $script->name);
+        $this->assertEquals('9dd04fae-d45d-5a64-a45d-62d14d2c62b5', $script->uuid);
+        $this->assertEquals(str_replace('stores/', '', $store->context), $script->store_hash);
+        $this->assertEquals($store->id, $script->store_id);
```

Comment:
These assertions appear quite often. You may want to create a helper that asserts all these values for you.